ui/lib/kv/addon/routes/secret/metadata.js

```diff
@@ -7,31 +7,63 @@ import Route from '@ember/routing/route';
 import { service } from '@ember/service';
 
 export default class KvSecretMetadataRoute extends Route {
-  @service store;
+  @service capabilities;
   @service secretMountPath;
+  @service store;
 
   fetchMetadata(backend, path) {
     return this.store.queryRecord('kv/metadata', { backend, path }).catch((error) => {
       if (error.message === 'Control Group encountered') {
         throw error;
       }
-      return {};
+      return null;
     });
   }
 
+  async fetchCapabilities(backend, path) {
```

Comment:
This is new, so i'm not sure how folks will feel about it. It could be generalized and a method added to the capabilities service but I wanted to stew on it some more 🍲